feat(agent): A2A Served Agent Support for Multi-Turn

njbrake opened the pull request:
To keep this PR manageable, note that this does not include any changes to the a2a-as-a-tool client side implementation, it is purely the server side.

Once we feel good about the server support, I'll make another PR to add support for a multi-turn agent-to-agent conversation using a2a-as-a-tool.

daavoo commented on src/any_agent/serving/task_manager.py:
```diff
@@ -0,0 +1,170 @@
+        self._tasks: dict[str, TaskData] = {}
+        self._last_cleanup: datetime | None = None
+
+    def _should_run_cleanup(self) -> bool:
```
Is this really needed? From a quick look it looks like potentially a premature optimization.
Like, how expensive is to always check if there are tasks and are expired vs always running this (which the increased code that we need to have for it)

daavoo commented on tests/integration/test_session_management.py:
```diff
@@ -0,0 +1,144 @@
+
+
+@pytest.mark.asyncio
+async def test_task_management_multi_turn_conversation():
```
Found the test very helpful to understand the behavior ❤️

daavoo commented on tests/integration/test_session_management.py:
```diff
@@ -0,0 +1,144 @@
+                    "taskId": response_1.root.result.id,
+                },
+            }
+            request_3 = SendMessageRequest(
```
Is it worth adding an extra step to test "overriding" information? Like. "ups forget what I said, I turned 31 last week"

njbrake replied:
Hmm. I don't think so? At that point its more testing out how the model responds to clarifying/changing info about the history, which seems more related to model performance and not as related to whether we are correctly giving it the past history. Let me know if you think the overriding info adds something necessary to testing whether or not the history is provided and the state managed.

daavoo replied:
makes sense

njbrake commented on src/any_agent/serving/config.py:
```diff
@@ -1,6 +1,43 @@
+HistoryFormatter = Callable[[list[AgentMessage], str], str]
+
+
+def default_history_formatter(messages: list[AgentMessage], current_query: str) -> str:
```
The reason I put it here was to make it more transparent to the developer what we do to format the history by default. By moving it into the TaskManager then it becomes slightly less transparent what our default behavior is imo. However, I'll take some time to think about before I move forward with the current structure 👍

njbrake commented:
With the test I added, i was able to mock the LLM call (since it's not necessary to verify the behavior I wanted). Now the test takes less than half a second 🎉 

![Screenshot 2025-06-18 at 10 48 28 AM](https://github.com/user-attachments/assets/00775b8e-a6f4-4290-b284-a20cee5d8b7b)

I decided to keep it as an integration test because I'm still actually serving the agent over a port, its just that the agent run call is mocked so we don't make calls to an LLM. We have other integration tests that test that run_async is working as expected so I don't bother re-testing that functionality in the test I added.
